torch/csrc/lazy/core/ir.h

```diff
@@ -188,6 +193,43 @@ const T* NodeCast(const Node* node, OpKind op) {
 #endif
 }
 
+template <typename T, typename... Args>
+NodePtr ReuseNode(OpKind op, Args&&... args) {
+  if (FLAGS_torch_lazy_reuse_ir) {
+    std::deque<std::unique_ptr<TrieNode>>& successors =
+        Trie::Get()->Current()->successors;
+    for (auto it = successors.begin(); it != successors.end(); it++) {
+      NodePtr ir_node = (*it)->ir_node;
```

Comment:
would it be easy to move this logic inside of a Trie:: class method (e.g. LookupNode) which accepts std::forward args, and internally handles the SetCurrent logic? I felt like exposing 'SetCurrent' seemed a little odd